### ISSUE TITLE
False-positive MagicNumber issue reported when using negative numbers

### ISSUE THREAD

**Issue by xrigau** [arturbosch/detekt#530]
It seems that detekt is reporting a false-positive `MagicNumber` issue when defining a negative number as a constant.

Code to reproduce:

```
package com.foo.bar

class Test {
    companion object {
        private const val FALSE_POSITIVE = -180
        private const val THIS_IS_FINE = 180
    }
}
```

Running `./gradlew detektCheck` reports the following issue:

```
Ruleset: style
        MagicNumber - [FALSE_POSITIVE] at com/foo/bar/Test.kt:12:43
```

Using the annotation as expected (without raising the issue):

```
    companion object {
        @Suppress("MagicNumber")
        private const val THIS_WORKS = -180
    }
```

Currently using version `1.0.0.RC4-3` but I also tried with the latest `1.0.0.RC5-3` and got the same results.
Relevant configuration:

```
  MagicNumber:
    active: true
    ignoreNumbers: '-1,0,1,2'
    ignoreHashCodeFunction: true
    ignorePropertyDeclaration: false
    ignoreAnnotation: true
```

I also tried adding extra options:

```
    ignoreCompanionObjectPropertyDeclaration: true
    ignoreConstantDeclaration: true
```

But it didn't make a difference

**Comment by rock3r** [comment]
This is probably hitting a corner case in the companion object/constants detection... It should work (and negative numbers [are covered](https://github.com/arturbosch/detekt/blob/master/detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumberSpec.kt#L98-L120) by the logic). Will maybe take a look at some point if I have some spare time over the next few days.

**Comment by arturbosch** [comment]
Just checked fast, but could not reproduce it

```kotlin
	given("false positive negative constant number") {
		it("test") {
			val code = """
				package com.foo.bar

				class Test {
					companion object {
						private const val FALSE_POSITIVE = -180
						private const val THIS_IS_FINE = 180
					}
				}
			"""
			assertThat(MagicNumber().lint(code)).isEmpty()
		}
	}
```

**Comment by arturbosch** [comment]
Feel free to reopen this if the issue still happening for you from a master build.

**Comment by xrigau** [comment]
sorry I've been AFK for a few days, will have a look again and reopen if the issue is still there, thanks!
